system/Validation/Rules.php

```diff
@@ -280,10 +280,10 @@ public function required_with($str = null, ?string $fields = null, array $data =
      *
      * @param string|null $str
      */
-    public function required_without($str = null, ?string $fields = null, array $data = []): bool
+    public function required_without($str = null, ?string $fields = null, array $data = [], ?string $keyField = null): bool
     {
-        if ($fields === null || empty($data)) {
-            throw new InvalidArgumentException('You must supply the parameters: fields, data.');
+        if ($fields === null || empty($data) || $keyField === null) {
+            throw new InvalidArgumentException('You must supply the parameters: fields, data, keyField.');
```

Comment:
Yes, it seems a breaking change.

```php
 $this->validation->setRules([
    'a.*.c' => 'required_without[a.*.b]',
])->run($data);
```
But I think this param is needed because this rule need to know what the key field is being passed in now (like: `a.0.c`, `a.1.c`...etc), otherwise it will only be passed the asterisk field (like: `a.*.c`).